### DISCUSSION TITLE
How to handle React elements and their crappy randomized classes?

### DISCUSSION THREAD

**Discussion by Gitoffthelawn** [darkreader/darkreader#8091] [Q&A]
Facebook's React (which is now being used by thousands of sites... ugh) is a big pain in the neck when trying to appropriately style elements for dark and light themes.

The issue is that React appends random data to class names for HTML elements.  Thus, you can't just specify an element by it's class like you're supposed to be able to do.  Frickin' Zuck and his stooges.

The best workaround I've come up with is to use `^=` when specifying class names in CSS (and then only include the non-randomized part of the class name).  Since the browser has to now do more work (it can't just quickly perform an `eq` on the strings), it results in reduced performance, especially on sites with a massive number of elements (which is pretty much every site that foolishly chooses to use React).  This is especially true when you have to do something ugly like `*[class^="outer_"] > *[class^="inner]` instead of just `.outer > .inner`, but it's the best workaround I've come up with so far.

Anyone come up with anything better?

**Comment by Gusted** [comment]
> Facebook's React (which is now being used by thousands of sites... ugh) is a big pain in the neck when trying to appropriately style elements for dark and light themes.
> 
> The issue is that React appends random data to class names for HTML elements. Thus, you can't just specify an element by it's class like you're supposed to be able to do. Frickin' Zuck and his stooges.

To clarify: react by default doesn't do this, but rather other frameworks that are built around this.



> Anyone come up with anything better?

From what I see, that's the best you could do, it really requires clever tricks to work around this(just see examples from `drive.google.com` they are spooky CSS which should once be crafted carefully, and then hope it should never be touched again.

**Reply by Gitoffthelawn** [reply]
> To clarify: react by default doesn't do this, but rather other frameworks that are built around this.

Thanks.  I did not know this (until now).  It looks like most React code in the wild uses additional frameworks because every time I have to deal with a crappy React site, I'm dealing with randomized class names.  Even Amazon is starting to use this crap, and it's making their slow performance site even more laggy.

Which React frameworks are most commonly used that generate these randomized class names?

> > Anyone come up with anything better?
> 
> From what I see, that's the best you could do, it really requires clever tricks to work around this(just see examples from `drive.google.com` they are spooky CSS which should once be crafted carefully, and then hope it should never be touched again.

Sounds good.  If you or anyone else, comes up with anything better, I'm all ~~ears~~ eyes.
